From 480e7d44bef3921ebdd9a4e14a2c0e9d112531bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 9 Mar 2020 17:43:54 +0100
Subject: [PATCH 02/48] screen-cast-stream-src: Don't complain when we can't
 dequeue buffer

PipeWire will be unable to dequeue a buffer if all are already busy.
This can happen for valid reasons, e.g. the stream consumer not being
fast enough, so don't complain in the journal if it happens.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1115
---
 src/backends/meta-screen-cast-stream-src.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/backends/meta-screen-cast-stream-src.c b/src/backends/meta-screen-cast-stream-src.c
index 9c3657c75..170f34043 100644
--- a/src/backends/meta-screen-cast-stream-src.c
+++ b/src/backends/meta-screen-cast-stream-src.c
@@ -456,10 +456,7 @@ meta_screen_cast_stream_src_maybe_record_frame (MetaScreenCastStreamSrc *src)
 
   buffer = pw_stream_dequeue_buffer (priv->pipewire_stream);
   if (!buffer)
-    {
-      g_warning ("Failed to dequeue at PipeWire buffer");
-      return;
-    }
+    return;
 
   spa_buffer = buffer->buffer;
   data = spa_buffer->datas[0].data;
-- 
2.26.0.rc2

